dumper/main.py

```diff
@@ -26,27 +29,53 @@ def dot():
     sys.stdout.flush()
 
 
+def dump_forms(query, writer):
+    counter = 0
+    unique_domains = set()
+    for doc in iter_forms_data(es, index, query):
+        counter += 1
+        if counter % 100 == 0:
+            dot()
+        form_obj = parse_form_doc(doc)
+        form_obj.write(writer)
+        unique_domains.add(form_obj.domain)
+
+    write_domains(unique_domains, args.domain_output)
+
+
+def dump_users(query, writer):
```

Comment:
The parsing part of this I did test by running it on the sample output you gave me